# Melbourne Housing Market

Dataset referente ao mercado imobiliario de Melbourne.(preciso melhorar a descrição)


Dataset encontra-se disponivel no endereço: https://www.kaggle.com/datasets/anthonypino/melbourne-housing-market

In [44]:
import pandas as pd
import numpy as np

In [45]:
from sklearn.linear_model import LinearRegression
ols = LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.impute import SimpleImputer
from sklearn.metrics import mean_absolute_error 
from sklearn.metrics import mean_squared_error 
from sklearn.metrics import r2_score 

In [46]:
df = pd.read_csv(r"C:\Users\dougl\OneDrive\Área de Trabalho\Projetos GitHub\Melbourne_house\archive\Melbourne_housing_FULL.csv")

In [47]:
df.head(5)

Suburb             Address  Rooms Type      Price Method SellerG  \
0  Abbotsford       68 Studley St      2    h        NaN     SS  Jellis   
1  Abbotsford        85 Turner St      2    h  1480000.0      S  Biggin   
2  Abbotsford     25 Bloomburg St      2    h  1035000.0      S  Biggin   
3  Abbotsford  18/659 Victoria St      3    u        NaN     VB  Rounds   
4  Abbotsford        5 Charles St      3    h  1465000.0     SP  Biggin   

        Date  Distance  Postcode  ...  Bathroom  Car  Landsize  BuildingArea  \
0  3/09/2016       2.5    3067.0  ...       1.0  1.0     126.0           NaN   
1  3/12/2016       2.5    3067.0  ...       1.0  1.0     202.0           NaN   
2  4/02/2016       2.5    3067.0  ...       1.0  0.0     156.0          79.0   
3  4/02/2016       2.5    3067.0  ...       2.0  1.0       0.0           NaN   
4  4/03/2017       2.5    3067.0  ...       2.0  0.0     134.0         150.0   

   YearBuilt         CouncilArea Lattitude  Longtitude             Regionname  \
0        NaN  Yarra City Council  -37.8014    144.9958  Northern Metropolitan   
1        NaN  Yarra City Council  -37.7996    144.9984  Northern Metropolitan   
2     1900.0  Yarra City Council  -37.8079    144.9934  Northern Metropolitan   
3        NaN  Yarra City Council  -37.8114    145.0116  Northern Metropolitan   
4     1900.0  Yarra City Council  -37.8093    144.9944  Northern Metropolitan   

  Propertycount  
0        4019.0  
1        4019.0  
2        4019.0  
3        4019.0  
4        4019.0  

[5 rows x 21 columns]

In [48]:
df.sample(5)

Suburb              Address  Rooms Type      Price Method  \
29502      Frankston      5 Finisterre Ct      4    h   814250.0      S   
34198         Eltham      2/137 Bolton St      3    h   672200.0     SP   
2897   Carlton North       354 Station St      3    h  1214000.0      S   
18047    Forest Hill  4/13 Glen Valley Rd      1    h   360000.0     SP   
2067   Brighton East         27 Arnold Rd      4    h  2200000.0     PI   

          SellerG        Date  Distance  Postcode  ...  Bathroom  Car  \
29502         Ray   6/01/2018      38.0    3199.0  ...       2.0  2.0   
34198  Buckingham  24/02/2018      18.0    3095.0  ...       NaN  NaN   
2897       Nelson  16/04/2016       3.2    3054.0  ...       3.0  1.0   
18047         Ray  12/08/2017      15.4    3131.0  ...       1.0  1.0   
2067       Buxton  18/06/2016      10.7    3187.0  ...       2.0  2.0   

       Landsize  BuildingArea  YearBuilt              CouncilArea Lattitude  \
29502     884.0           NaN        NaN   Frankston City Council -38.16927   
34198       NaN           NaN        NaN     Banyule City Council       NaN   
2897      144.0           NaN        NaN   Melbourne City Council -37.78850   
18047      76.0           NaN     1970.0  Whitehorse City Council -37.83221   
2067      654.0         286.0     2000.0     Bayside City Council -37.92920   

       Longtitude                  Regionname Propertycount  
29502   145.15824  South-Eastern Metropolitan       17055.0  
34198         NaN        Eastern Metropolitan        6990.0  
2897    144.97570       Northern Metropolitan        3106.0  
18047   145.17117        Eastern Metropolitan        4385.0  
2067    145.02360       Southern Metropolitan        6938.0  

[5 rows x 21 columns]

In [49]:
df.tail(5)

Suburb           Address  Rooms Type      Price Method  \
34852  Yarraville       13 Burns St      4    h  1480000.0     PI   
34853  Yarraville     29A Murray St      2    h   888000.0     SP   
34854  Yarraville    147A Severn St      2    t   705000.0      S   
34855  Yarraville  12/37 Stephen St      3    h  1140000.0     SP   
34856  Yarraville  3 Tarrengower St      2    h  1020000.0     PI   

             SellerG        Date  Distance  Postcode  ...  Bathroom  Car  \
34852            Jas  24/02/2018       6.3    3013.0  ...       1.0  3.0   
34853        Sweeney  24/02/2018       6.3    3013.0  ...       2.0  1.0   
34854            Jas  24/02/2018       6.3    3013.0  ...       1.0  2.0   
34855  hockingstuart  24/02/2018       6.3    3013.0  ...       NaN  NaN   
34856             RW  24/02/2018       6.3    3013.0  ...       1.0  0.0   

       Landsize  BuildingArea  YearBuilt               CouncilArea Lattitude  \
34852     593.0           NaN        NaN  Maribyrnong City Council -37.81053   
34853      98.0         104.0     2018.0  Maribyrnong City Council -37.81551   
34854     220.0         120.0     2000.0  Maribyrnong City Council -37.82286   
34855       NaN           NaN        NaN  Maribyrnong City Council       NaN   
34856     250.0         103.0     1930.0  Maribyrnong City Council -37.81810   

       Longtitude            Regionname Propertycount  
34852   144.88467  Western Metropolitan        6543.0  
34853   144.88826  Western Metropolitan        6543.0  
34854   144.87856  Western Metropolitan        6543.0  
34855         NaN  Western Metropolitan        6543.0  
34856   144.89351  Western Metropolitan        6543.0  

[5 rows x 21 columns]

In [50]:
df.shape

(34857, 21)

In [51]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34857 entries, 0 to 34856
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Suburb         34857 non-null  object 
 1   Address        34857 non-null  object 
 2   Rooms          34857 non-null  int64  
 3   Type           34857 non-null  object 
 4   Price          27247 non-null  float64
 5   Method         34857 non-null  object 
 6   SellerG        34857 non-null  object 
 7   Date           34857 non-null  object 
 8   Distance       34856 non-null  float64
 9   Postcode       34856 non-null  float64
 10  Bedroom2       26640 non-null  float64
 11  Bathroom       26631 non-null  float64
 12  Car            26129 non-null  float64
 13  Landsize       23047 non-null  float64
 14  BuildingArea   13742 non-null  float64
 15  YearBuilt      15551 non-null  float64
 16  CouncilArea    34854 non-null  object 
 17  Lattitude      26881 non-null  float64
 18  Longti

In [52]:
df.duplicated().sum()

1

In [53]:
df.isnull().sum()

Suburb               0
Address              0
Rooms                0
Type                 0
Price             7610
Method               0
SellerG              0
Date                 0
Distance             1
Postcode             1
Bedroom2          8217
Bathroom          8226
Car               8728
Landsize         11810
BuildingArea     21115
YearBuilt        19306
CouncilArea          3
Lattitude         7976
Longtitude        7976
Regionname           3
Propertycount        3
dtype: int64

É possivel visualizar acima, os dados que apresentão algum tipo de campo nulo. será iniado o tratamento apartir destes dados.

Para estimar um valor, é necessario que alguns campos tenha obrigatoriamente valores. Caso não tenha, esse dado sera descartado.
* Bedroom2 possui 8217 campos nulos.
* Bathroom possui 8226 campos nulos.
* Price possui 7610 campos nulos.


In [54]:
#df.dropna(subset=["Bedroom2"],inplace=True)
#df.dropna(subset=["Bathroom"],inplace=True)
#df.dropna(subset=["Price"],inplace=True)
#df.dropna(subset=["BuildingArea"],inplace=True)


In [55]:
df.isnull().sum()

Suburb               0
Address              0
Rooms                0
Type                 0
Price             7610
Method               0
SellerG              0
Date                 0
Distance             1
Postcode             1
Bedroom2          8217
Bathroom          8226
Car               8728
Landsize         11810
BuildingArea     21115
YearBuilt        19306
CouncilArea          3
Lattitude         7976
Longtitude        7976
Regionname           3
Propertycount        3
dtype: int64

In [56]:
df["Car"].value_counts()

2.0     12214
1.0      9164
0.0      1631
3.0      1606
4.0      1161
5.0       151
6.0       140
7.0        25
8.0        23
10.0        6
9.0         3
11.0        2
26.0        1
18.0        1
12.0        1
Name: Car, dtype: int64

In [57]:
df[df["CouncilArea"].isnull()]

Suburb           Address  Rooms Type     Price Method SellerG  \
18523    Footscray  2/16 Stafford St      2    u  710000.0      S     Jas   
26888   Camberwell   2/3 Kingsley St      2    h  825000.0     VB  Jellis   
29483  Fawkner Lot      1/3 Brian St      3    h  616000.0     SP    Brad   

             Date  Distance  Postcode  ...  Bathroom  Car  Landsize  \
18523  15/07/2017       5.1    3011.0  ...       NaN  NaN       NaN   
26888  11/11/2017       7.7    3124.0  ...       NaN  NaN       NaN   
29483   6/01/2018       NaN       NaN  ...       NaN  NaN       NaN   

       BuildingArea  YearBuilt  CouncilArea Lattitude  Longtitude  Regionname  \
18523           NaN        NaN          NaN       NaN         NaN         NaN   
26888           NaN        NaN          NaN       NaN         NaN         NaN   
29483           NaN        NaN          NaN       NaN         NaN         NaN   

      Propertycount  
18523           NaN  
26888           NaN  
29483           NaN  

[3 rows x 21 columns]

Nao fazer alteraçoes abaixo

In [58]:
X = df.drop(["Price","Car", "Suburb","Distance","Postcode","Longtitude", "Propertycount","Lattitude", "Address", "Type", "Method", "YearBuilt", "SellerG", "BuildingArea", "Date", "CouncilArea", "Regionname"], axis=1)

In [59]:
Y = df["Price"]

In [60]:
X

Rooms  Bedroom2  Bathroom  Landsize
0          2       2.0       1.0     126.0
1          2       2.0       1.0     202.0
2          2       2.0       1.0     156.0
3          3       3.0       2.0       0.0
4          3       3.0       2.0     134.0
...      ...       ...       ...       ...
34852      4       4.0       1.0     593.0
34853      2       2.0       2.0      98.0
34854      2       2.0       1.0     220.0
34855      3       NaN       NaN       NaN
34856      2       2.0       1.0     250.0

[34857 rows x 4 columns]

In [61]:
X.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 34857 entries, 0 to 34856
Data columns (total 4 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   Rooms     34857 non-null  int64  
 1   Bedroom2  26640 non-null  float64
 2   Bathroom  26631 non-null  float64
 3   Landsize  23047 non-null  float64
dtypes: float64(3), int64(1)
memory usage: 1.1 MB


In [62]:
Y

0              NaN
1        1480000.0
2        1035000.0
3              NaN
4        1465000.0
           ...    
34852    1480000.0
34853     888000.0
34854     705000.0
34855    1140000.0
34856    1020000.0
Name: Price, Length: 34857, dtype: float64

In [63]:
Y.info()

<class 'pandas.core.series.Series'>
RangeIndex: 34857 entries, 0 to 34856
Series name: Price
Non-Null Count  Dtype  
--------------  -----  
27247 non-null  float64
dtypes: float64(1)
memory usage: 272.4 KB


In [64]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.28, random_state=42)

NameError: name 'y' is not defined

In [ ]:
X.shape, y.shape, X_train.shape, X_test.shape, y_train.shape, y_test.shape

((13580, 4), (13580,), (9777, 4), (3803, 4), (9777,), (3803,))

In [ ]:
ols.fit(X,Y)

LinearRegression()

In [ ]:
ols.intercept_

31584.605965678697

In [ ]:
ols.coef_

array([2.24216833e+05, 2.53683552e+03, 2.45242799e+05, 3.03574181e+00])

In [ ]:
predito = ols.predict(X)
predito

array([ 730947.96150257,  730808.31737929, 1202737.99820381, ...,
       1203654.79223056, 1186471.03115835, 1184941.0172859 ])

In [ ]:
mean_absolute_error(y, predito) 

377922.38368053216

In [ ]:
mean_squared_error(y, predito)

288970799009.69965

In [ ]:
r2_score(y, predito)

0.29293073279582105